src/ProxyClient/Varnish.php

```diff
@@ -125,28 +125,36 @@ public function refresh($url, array $headers = array())
 
         return $this;
     }
-
-    /**
-     * {@inheritdoc}
-     *
-     * @throws MissingHostException If a relative path is queued for purge/
-     *                              refresh and no base URL is set
-     *
-     */
-    protected function createRequest($method, $url, array $headers = array())
+    
+    protected function queueRequest($method, $url, array $headers = array())
     {
-        $request = parent::createRequest($method, $url, $headers);
-
-        // For purge and refresh, add a host header to the request if it hasn't
-        // been set
-        if (self::HTTP_METHOD_BAN !== $method
-            && '' == $request->getHeader('Host')
-        ) {
-            throw MissingHostException::missingHost($url);
+        if (self::HTTP_METHOD_BAN !== $method) {
+            // AND NO HOST IS SET
```

Comment:
can we validate that either the constructor gets a host name/ip or $url contains one? even when purging 127.0.0.1 we should require this to be explicit?